dsl/src/commonMain/kotlin/it/unibo/collektive/SharingContext.kt

```diff
@@ -0,0 +1,29 @@
+package it.unibo.collektive
+
+/**
+ * The lambda context passed to the [share] function.
+ */
+class SharingContext<Initial, Return> {
+    internal var toBeSent: Initial? = null
+    internal var toBeReturned: Return? = null
+    internal var areSameType = true
+
+    /**
+     * Express the value [toReturn] after the [share] computation is done.
+     * It can be used with checks after the invocation.
+     *
+     * ## Example
+     * ```
+     * val res3 = share(initV1) {
```

Comment:
Where `initV1` come from? Use a different initial value.
In the examples, a primitive type would be better.